packages/@o3r/schematics/src/utility/ast.ts

```diff
@@ -87,3 +87,135 @@ export function getExportedSymbolsFromFile(program: ts.Program, sourcePath: stri
 
   return typeChecker.getExportsOfModule(symbol!);
 }
+
+export type DecoratorWithArg = ts.Decorator & {
+  expression: ts.CallExpression & {
+    expression: ts.Identifier;
+  };
+};
+
+/**
+ * Returns true if it is a decorator with arguments
+ *
+ * @param node decorator node
+ */
+export const isDecoratorWithArg = (node: ts.Node): node is DecoratorWithArg =>
+  ts.isDecorator(node)
+  && ts.isCallExpression(node.expression)
+  && ts.isIdentifier(node.expression.expression);
+
+/**
+ * Returns the value of `argName` in the first argument
+ *
+ * @param decorator
+ * @param argName
+ */
+export const getPropertyFromDecoratorFirstArgument = (decorator: DecoratorWithArg, argName: string) => {
+  const ngClassDecoratorArgument = decorator.expression.arguments[0];
+
+  if (ts.isObjectLiteralExpression(ngClassDecoratorArgument)) {
+    return ngClassDecoratorArgument.properties.find((prop): prop is ts.PropertyAssignment =>
+      ts.isPropertyAssignment(prop)
+      && ts.isIdentifier(prop.name)
+      && prop.name.escapedText.toString() === argName
+    )?.initializer;
+  }
+};
+
+/**
+ * Returns `ExpressionWithTypeArguments[]` of a class that implements `str`
+ *
+ * @param str
+ */
+export const getImplementsExpressionWithTypeArguments = (str: string) => {
+  const sourceFile = ts.createSourceFile(
+    'index.ts',
+    `class A implements ${str} {}`,
+    ts.ScriptTarget.ES2020,
+    true
+  );
+  return [...(sourceFile.statements[0] as ts.ClassDeclaration).heritageClauses![0].types];
+};
+
+/**
+ * Returns `ClassElement[]` of a class that have `str` has body
+ *
+ * @param str
+ */
+export const getClassElementsFromString = (str: string) => {
```

Comment:
For the similar function, it is actually more `generate` (or `create`) than `get` that should be used as prefix.